projects/core/src/site-context/config/config-loader/site-context-config-loader.service.ts

```diff
@@ -0,0 +1,144 @@
+import { DOCUMENT, isPlatformBrowser } from '@angular/common';
+import {
+  Inject,
+  Injectable,
+  isDevMode,
+  Optional,
+  PLATFORM_ID,
+} from '@angular/core';
+import { filter, map, take } from 'rxjs/operators';
+import { BaseSite } from '../../../model/misc.model';
+import { SERVER_REQUEST_URL } from '../../../util/ssr.tokens';
+import { BaseSiteService } from '../../facade/base-site.service';
+import {
+  BASE_SITE_CONTEXT_ID,
+  CURRENCY_CONTEXT_ID,
+  LANGUAGE_CONTEXT_ID,
+  THEME_CONTEXT_ID,
+} from '../../providers/context-ids';
+import { SiteContextConfig } from '../site-context-config';
+
+@Injectable({ providedIn: 'root' })
+export class SiteContextConfigLoaderService {
+  constructor(
+    protected baseSiteService: BaseSiteService,
+    @Inject(PLATFORM_ID) protected platform?: any,
+    @Inject(DOCUMENT) protected document?: any,
+    @Optional()
+    @Inject(SERVER_REQUEST_URL)
+    protected serverRequestUrl?: string
+  ) {}
+
+  private get currentUrl(): string | undefined {
+    if (isPlatformBrowser(this.platform)) {
+      return this.document.location.href;
+    }
+    if (this.serverRequestUrl) {
+      return this.serverRequestUrl;
+    }
+    if (isDevMode()) {
+      console.error(
+        `Please provide token 'SERVER_REQUEST_URL' with the requested URL for SSR`
+      );
+    }
+  }
+
+  /**
+   * Returns the site context config basing on the current base site data
+   */
+  loadConfig(): Promise<SiteContextConfig> {
```

Comment:
Would you mind changing the return type to Observable instead of Promise?

I'm aware it was already a Promise in the old class and now we're just moving the code to a new class.
At the same time, recently we agreed with the Blamed Team that for consistency our public API should return Observables rather than Promises (even when they complete after the first emission).

Btw. then you might need moving `.toPromise()` to the consumer - to the [config initializer](https://github.com/SAP/spartacus/pull/11469/files#diff-48696391b9b48939682bd282f62570149e154ea59a3969676734bae57f25053fR33)

Reply:
I don't mind :). Actually I like Observable more like Promise. But, since we use Promise in that old OccConfigLoaderService, so I use Promise here as well.

Reply:
Done